pkg/api/serialization_test.go

```diff
@@ -86,14 +86,22 @@ func roundTrip(t *testing.T, codec runtime.Codec, item runtime.Object) {
 }
 
 // roundTripSame verifies the same source object is tested in all API versions.
-func roundTripSame(t *testing.T, item runtime.Object, except ...string) {
+func roundTripSame(t *testing.T, item runtime.Object, kind string, except ...string) {
 	set := util.NewStringSet(except...)
 	seed := rand.Int63()
 	fuzzInternalObject(t, "", item, seed)
+
 	version := testapi.Version()
+	codec := testapi.Codec()
+	//TODO: we should detect if a kind belongs to "experimental" group once we
+	//divide the scheme for internal objects.
+	if api.Scheme.Recognizes(testapi.GroupAndVersion("experimental"), kind) {
```

Comment:
I guess the point of this is to get the experimental group version if the object is in that group? Can you add another TODO, as this will not work at all once we can have N groups.